### PR TITLE
Fix table of contents and a few typos

### PR DESCRIPTION
**Pull request by t-bast**
I messed up the table of contents with my last commit, here is the fix.
I also took this opportunity to:

- add the `channel_type` tlv
- fix a few typos / markdown formatter issues
- review some of the Bolt 3 changes

My only two remaining comments are:

- I think we should be more aggressive than 65/64 when doing RBF: what about mandating a 10% increase (or another %, but more aggressive than 1/64) to avoid needlessly spamming the bitcoin p2p network?
- The Bolt 3 appendix G (dual funding test vectors) should be updated, I'll be able to verify them once they're complete

The eclair implementation is getting close, I haven't implemented the RBF part yet but apart from that it's almost complete.

### CODE REVIEW

File: 03-transactions.md [hunk 1]
```diff
@@ -724,37 +724,37 @@ at each bucket is a prefix of the desired index.
...
-        //                       + sum(scriptlen) + sum(var_ints) +
-	funding_transaction_weight = 4 * (funding_transaction + transaction_fields) + segwit_fields
+      // transaction_fields = 10 (version, input count, output count, locktime)
+      // segwit_fields = 2 (marker + flag)
+      // funding_transaction = 43 + num_inputs * 40 + num_outputs * 8
+      //                       + sum(scriptlen) + sum(var_ints) +
+      funding_transaction_weight = 4 * (funding_transaction + transaction_fields) + segwit_fields
 
-	witness_weight = sum(max_witness_len)
+      witness_weight = sum(witness_len)
```
Comment (t-bast) [inline 03-transactions.md:755]: @niftynei can you explain why you put `max_witness_len` here? What did this refer to?

**Comment by niftynei** [comment]
> I think we should be more aggressive than 65/64 when doing RBF: what about mandating a 10% increase (or another %, but more aggressive than 1/64) to avoid needlessly spamming the bitcoin p2p network?

Sure, how does 25/24 sound? That's 4.1% increase each time. Using the "[rule of 72](https://www.investopedia.com/terms/r/ruleof72.asp)", that doubles every 17 attempts; with 65/64 we double every ~45 attempts.

**Comment by t-bast** [comment]
> Sure, how does 25/24 sound?

Sounds good to me! Do you want me to update this PR to reflect that?

File: 03-transactions.md (hunk 1 above)
Comment (niftynei) [reply 03-transactions.md:755]: > The minimum witness weight for an input is 110.

It's a left over from before the `minimum witness weight` (or `minimum fee`). Originally we had each side send a `max_witness_len` in the `tx_add_input` message. Will update.

**Comment by niftynei** [comment]
> Do you want me to update this PR to reflect that?

Yeah, that'd be great!

**Comment by t-bast** [comment]
Done in https://github.com/niftynei/lightning-rfc/pull/4/commits/98a813ea63c66695a0de8d9ebdc4154133e45179

**Pull request merged**